packages/compiler/core/checker.ts

```diff
@@ -4588,7 +4588,7 @@ export function createChecker(program: Program): Checker {
   function areScalarsRelated(source: Scalar, target: Scalar) {
     let current: Scalar | undefined = source;
     while (current) {
-      if (current === target) {
+      if (current.name === target.name) {
```

Comment:
@nguerrera @bterlson this is the change that I think may be controversial. Without this, it sees the two instances of "string" in the test as different and not equivalent.

Reply:
I will need to pull this down and debug. I can do that in the morning.

Reply:
Looking now

Reply:
Confirming repro:

```
Z:/test/main.cadl:3:42 - error unassignable: Type 'Cadl.string' is not assignable to type 'Cadl.string'
```

Reply:
It appears the template arg has not been projected while the property type has been.

Reply:
@nguerrera is that a bug fix in the projection system I should wait for or do you think my `.name` comparison is fine?

Reply:
I'm working on a fix, please wait for that PR and then revert the name check

Reply:
https://github.com/microsoft/cadl/pull/1648

Mind trying to apply that fix locally and confirming it resolves the issue?

Reply:
I pulled in your change and was able to revert this. Thanks for the debugging and the fix @nguerrera!!!